src/professionals/professionals.service.ts

```diff
@@ -1,79 +1,53 @@
 import { Injectable } from '@nestjs/common';
-
-const users = [
-  {
-    id: 1,
-    name: 'Leanne Graham',
-    username: 'Bret',
-    email: 'Sincere@april.biz',
-    address: {
-      street: 'Lucas Light',
-      suite: 'Apt. 556',
-      city: 'Montevideo',
-      zip: '92998-3874',
-      geo: {
-        lat: '-37.3159',
-        lng: '81.1496',
-      },
-    },
-    phone: '1-770-736-8031 x56442',
-    website: 'hildegard.org',
-    company: {
-      name: 'Pedro',
-      catchPhrase: 'Multi-layered client-server neural-net',
-      bs: 'harness real-time e-markets',
-    },
-  },
-  {
-    id: 2,
-    name: 'Ervin Howell',
-    username: 'lucas',
-    email: 'Shanna@melissa.tv',
-    address: {
-      street: 'Victor Plains',
-      suite: 'Suite 879',
-      city: 'Montevideo',
-      zip: '90566-7771',
-      geo: {
-        lat: '-43.9509',
-        lng: '-34.4618',
-      },
-    },
-    phone: '010-692-6593 x09125',
-    website: 'anastasia.net',
-    company: {
-      name: 'Microsoft',
-      catchPhrase: 'Proactive didactic contingency',
-      bs: 'supply scalable supply-chains',
-    },
-  },
-];
+import { NotFoundException } from '@nestjs/common/exceptions';
+import { Req } from 'src/dtos';
+import { ProfessionalEntity } from 'src/entities';
+import { ProfessionalsRepository } from '../repositories/professionals.repository';
 
 @Injectable()
 export class ProfessionalsService {
-  public async getMany(): Promise<any> {
-    return users;
+  constructor(private professionalsRepository: ProfessionalsRepository) {}
+
+  public async getProfessionals(
+    professionalsFilterDto: Req.GetProfessionalsFilterDto,
+  ): Promise<ProfessionalEntity[]> {
+    return this.professionalsRepository.getProfessionals(
+      professionalsFilterDto,
+    );
   }
 
-  public async getOne(id: number): Promise<any> {
-    return users.find((user) => user.id === id);
+  public async getProfessionalById(id: string): Promise<ProfessionalEntity> {
+    const professional = await this.professionalsRepository.findOneBy({ id });
+
+    if (!professional) {
+      throw new NotFoundException(`Professional with ID "${id}" not found`);
+    }
+
+    return professional;
```

Comment:
```suggestion
    const professional = await this.professionalsRepository.findOneBy({ id });

    if (!professional) {
      throw new NotFoundException(`Professional with ID "${id}" not found`);
    }

    return professional;
```

There's `findOneOrFail` which raises and Error if no entities are found. Using that allows you to let the servicer layer decide if it should handle and how to handle such error.

Reply:
Didn't know about it, updated it to use `findOneByOrFail`.